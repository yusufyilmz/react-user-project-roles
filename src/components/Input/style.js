
import styled from 'styled-components';

export const Label = styled.label`
    float: left;
    width: 25%;
    margin: 15px 5px;
    font-size: x-large;
    `

Label.displayName = 'Label'


export const Input = styled.input`
    margin-left: auto;
    margin-right: auto;
    width: 100%;
    border: 1px solid #ccc;
    border-radius: 4px;
    resize: vertical;
    background-color: ${props => (props.type === 'submit' || props.type === 'button') && '#27435c'};
    display: ${props => (props.type === 'submit' || props.type === 'button') && 'flex'};
    justify-content: ${props => (props.type === 'submit' || props.type === 'button') && 'center'};
    color: ${props => (props.type === 'submit' || props.type === 'button') && 'white'};
    height: ${props => (props.type === 'submit' || props.type === 'button') && '50px'};
    text-indent: ${props => (props.type === 'text' ) && '30px'};
    font-size: ${props => (props.type === 'submit' || props.type === 'button') ? 'x-large' : 'medium' };
    padding: ${props => (props.type === 'submit' || props.type === 'button') ? '' : '15px 0' };

    `;

Input.displayName = 'Input'