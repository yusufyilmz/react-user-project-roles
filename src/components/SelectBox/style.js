

// import styled from 'styled-components';

// export const Ul = styled.ul`
// z-index: 10;
// position: absolute;
// width: 70%;
// border: 1px solid #dfdfdf;
//  border-bottom-right-radius: 3px; 
//  border-bottom-left-radius: 3px; 
// background-color: #fff;
// -webkit-box-shadow: 0 2px 5px -1px #e8e8e8;
// box-shadow: 0 2px 5px -1px #e8e8e8;
// font-weight: 700;
// max-height: 215px;
// overflow-y: scroll;
// -webkit-overflow-scrolling: touch;
//   `;

// export const Li = styled.li`
//     width: 100%;
//     font-size: 1rem;
//     padding: 8px 10px;
//     line-height: 1.6rem;
//     cursor: default;
//     display: inline-block;
//     white-space: nowrap;
//     text-overflow: ellipsis;
//   &.selected {
//     color: #fff;
//     background-color: #27435c;
//   }
//   &:hover {
//     color: white;
//     background-color: #27435c;
//   }`;






import styled from 'styled-components';

export const Ul = styled.ul`
    z-index: 10;
    position: absolute;
    width: 57%;
    margin-top: 0px;
    border: 1px solid #dfdfdf;
    border-bottom-right-radius: 3px;
    border-bottom-left-radius: 3px;
    background-color: #fff;
    -webkit-box-shadow: 0 2px 5px -1px #e8e8e8;
    box-shadow: 0 2px 5px -1px #e8e8e8;
    font-weight: 700;
    padding: 15px 0;
    max-height: 215px;
    overflow-y: scroll;
    -webkit-overflow-scrolling: touch;
    @media screen and (max-width: 600px) {
      width: 75%;
    }`;

export const Li = styled.li`
    width: 100%;
    font-size: 1rem;
    padding: 8px 10px;
    line-height: 1.6rem;
    cursor: default;
    display: inline-block;
    white-space: nowrap;
    text-overflow: ellipsis;
    &:selected {
      color: #fff;
      background-color: #27435c;
    }
    &:hover {
      color: black;
      background-color: #27435c;
    }`;
