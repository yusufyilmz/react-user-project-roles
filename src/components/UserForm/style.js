

import styled from 'styled-components';

export const UserFormWrapper = styled.form`
    padding: 10px;
`;

UserFormWrapper.displayName = 'Form'