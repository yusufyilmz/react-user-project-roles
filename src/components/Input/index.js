import React, { Fragment } from 'react';
import { Input, Label } from './style';


export default ({ type, name, text, onChange, value, onClick, required, placeHolder, width}) => {
    return (
        <Fragment>
            <Label htmlFor={name} >{text}</Label>
            <Input
                placeholder={placeHolder}
                width={width}
                name={name}
                autoComplete="off"
                required={required}
                id={name}
                type={type}
                onChange={onChange}
                onClick={onClick}
                value={value}>
            </Input>
        </Fragment>
    )
}


