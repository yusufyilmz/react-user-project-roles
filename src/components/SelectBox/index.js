import React, { Component } from 'react';
import { Ul, Li } from './style';
import Input from '../Input';
import onClickOutside from 'react-onclickoutside'


class SelectBox extends Component {

    constructor(props) {
        super(props);
        this.state = {
            listOpen: false,
            value: '',
            list: this.props.list,
        }
    }

    componentDidUpdate(prevProps) {
        if (prevProps.list !== this.props.list) {
            this.setState({ list: this.props.list })
        }
    }

    toggleList = (e) => {
        this.setState(prevState => ({
            listOpen: !prevState.listOpen,
            list: this.props.list,
            value: ''
        }))
    }

    selectItem(item) {
        this.setState(prevState => ({
            value: item.name,
            listOpen: !prevState.listOpen
        }), () => {
            this.props.selectItem(item);
        })
    }

    filterList = (e) => {
        const text = e.target.value;
        var list = this.props.list.filter(x => x.name.toLowerCase().startsWith(text.toLowerCase()))
        this.setState(prevState => ({
            value: text,
            list,
            listOpen: list.length > 0
        }), () => {
            this.props.selectItem({ name: text });

        })
    }
    handleClickOutside = (e) => {
        this.setState(prevState => ({
            listOpen: false
        }))
    }

    render() {
        return (
            <div>
                <Input
                    placeHolder={this.props.placeHolder}
                    text={this.props.text}
                    name={this.props.name}
                    onChange={this.filterList}
                    type="text"
                    required={true}
                    value={this.state.value}
                    onClick={this.toggleList} />
                {this.state.listOpen && <Ul >
                    {this.state.list.map((item) => {
                        if (item) return <Li
                            key={item.id}
                            onClick={() => this.selectItem(item)}>
                            {item.name}
                        </Li>
                    })}
                </Ul>
                }
            </div>
        )
    }

}

export default onClickOutside(SelectBox);
