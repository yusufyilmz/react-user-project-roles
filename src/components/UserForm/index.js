import React from 'react';
import SelectBox from '../SelectBox';
import UserFormContainer from '../../containers/userFormContainer';
import Input from '../Input';
import { UserFormWrapper } from './style';

const UserForm = () => {
    return (
        <UserFormContainer>
            {
                ({ roles, selectRole, projects, selectProject, users, selectUser, attendUser, userHasRole, removeRoleFromProject }) => (
                    <UserFormWrapper onSubmit={attendUser}>
                        <SelectBox
                            name="user"
                            text="User"
                            placeHolder="Enter or select username"
                            list={users}
                            selectItem={selectUser}
                            required={true} />
                        <SelectBox
                            name="role"
                            text="Roles"
                            placeHolder="Select role"
                            list={roles}
                            selectItem={selectRole}
                            required={true} />
                        <SelectBox
                            name="project"
                            text="Projects"
                            placeHolder="Select project"
                            list={projects}
                            selectItem={selectProject}
                            required={true} />
                        <Input
                            type="submit"
                            value="Submit role" />
                        {userHasRole &&
                            <Input
                                type="button"
                                value="Remove role of user"
                                onClick={removeRoleFromProject} />}
                    </UserFormWrapper>
                )
            }
        </UserFormContainer>
    );
}

export default UserForm;