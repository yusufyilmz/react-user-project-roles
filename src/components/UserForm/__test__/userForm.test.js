

import React from 'react';
import UserForm from '../index';
import Input from '../../Input';
import SelectBox from '../../SelectBox';
import { mount, shallow } from 'enzyme'
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

import ReduxThunkMiddleware from 'redux-thunk';
import ReduxPromiseMiddleware from 'redux-promise';

let middlewares = [ReduxThunkMiddleware, ReduxPromiseMiddleware]

describe('USERFORM COMPONENT', () => {

    const mockStore = configureMockStore(middlewares);
    const store = mockStore({
        form: {},
    });

    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <Provider store={store}>
                <UserForm/>
            </Provider>
        );
    });

    test('renders element correctly', () => {
        expect(wrapper.find('Form').at(0).length).toBe(1)
    })

    test('renders form element correctly', () => {
        expect(wrapper.find(SelectBox).length).toBe(3)
    })

    test('renders form element correctly', () => {
        expect(wrapper.find(Input).length).toBe(4)
        expect(wrapper.find('Label').length).toBe(4)
    })

    test('snapchat correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})
