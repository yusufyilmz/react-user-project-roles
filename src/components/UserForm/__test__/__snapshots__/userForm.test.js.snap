// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm component snapchat correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UserForm>
    <Connect(UserFormContainer)>
      <UserFormContainer
        addProjecRoleToUser={[Function]}
        fetchUserProjectsAndRoles={[Function]}
        removeRoleFromProject={[Function]}
        resetRemoveRoleFromProject={[Function]}
      >
        <Form
          onSubmit={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": true,
                  "lastClassName": "eoDtmv",
                  "rules": Array [
                    "
    padding: 10px;
",
                  ],
                },
                "displayName": "Form",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "form",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onSubmit={[Function]}
          >
            <form
              className="sc-ifAKCX eoDtmv"
              onSubmit={[Function]}
            >
              <OnClickOutside(SelectBox)
                eventTypes={
                  Array [
                    "mousedown",
                    "touchstart",
                  ]
                }
                excludeScrollbar={false}
                name="user"
                outsideClickIgnoreClass="ignore-react-onclickoutside"
                placeHolder="Enter or select username"
                preventDefault={false}
                required={true}
                selectItem={[Function]}
                stopPropagation={false}
                text="User"
              >
                <SelectBox
                  disableOnClickOutside={[Function]}
                  enableOnClickOutside={[Function]}
                  eventTypes={
                    Array [
                      "mousedown",
                      "touchstart",
                    ]
                  }
                  name="user"
                  outsideClickIgnoreClass="ignore-react-onclickoutside"
                  placeHolder="Enter or select username"
                  preventDefault={false}
                  required={true}
                  selectItem={[Function]}
                  stopPropagation={false}
                  text="User"
                >
                  <div>
                    <Component
                      name="user"
                      onChange={[Function]}
                      onClick={[Function]}
                      placeHolder="Enter or select username"
                      required={true}
                      text="User"
                      type="text"
                      value=""
                    >
                      <Label
                        htmlFor="user"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": true,
                                "lastClassName": "hApbuX",
                                "rules": Array [
                                  "
    float: left;
    width: 25%;
    margin: 15px 5px;",
                                ],
                              },
                              "displayName": "Label",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "label",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          htmlFor="user"
                        >
                          <label
                            className="sc-htpNat hApbuX"
                            htmlFor="user"
                          >
                            User
                          </label>
                        </StyledComponent>
                      </Label>
                      <Input
                        autoComplete="off"
                        id="user"
                        name="user"
                        onChange={[Function]}
                        onClick={[Function]}
                        placeholder="Enter or select username"
                        required={true}
                        type="text"
                        value=""
                      >
                        <StyledComponent
                          autoComplete="off"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bxivhb",
                                "isStatic": false,
                                "lastClassName": "byeNva",
                                "rules": Array [
                                  "
    margin-left: auto;
    margin-right: auto;
    width: 95%;
    padding: 12px;
    border: 1px solid #ccc;
    border-radius: 4px;
    resize: vertical;
    background-color: ",
                                  [Function],
                                  "
    display: ",
                                  [Function],
                                  "
    justify-content: ",
                                  [Function],
                                  "
    color: ",
                                  [Function],
                                  "
    ",
                                ],
                              },
                              "displayName": "Input",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bxivhb",
                              "target": "input",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          id="user"
                          name="user"
                          onChange={[Function]}
                          onClick={[Function]}
                          placeholder="Enter or select username"
                          required={true}
                          type="text"
                          value=""
                        >
                          <input
                            autoComplete="off"
                            className="sc-bxivhb beFuQG"
                            id="user"
                            name="user"
                            onChange={[Function]}
                            onClick={[Function]}
                            placeholder="Enter or select username"
                            required={true}
                            type="text"
                            value=""
                          />
                        </StyledComponent>
                      </Input>
                    </Component>
                  </div>
                </SelectBox>
              </OnClickOutside(SelectBox)>
              <OnClickOutside(SelectBox)
                eventTypes={
                  Array [
                    "mousedown",
                    "touchstart",
                  ]
                }
                excludeScrollbar={false}
                name="role"
                outsideClickIgnoreClass="ignore-react-onclickoutside"
                placeHolder="Select role"
                preventDefault={false}
                required={true}
                selectItem={[Function]}
                stopPropagation={false}
                text="Roles"
              >
                <SelectBox
                  disableOnClickOutside={[Function]}
                  enableOnClickOutside={[Function]}
                  eventTypes={
                    Array [
                      "mousedown",
                      "touchstart",
                    ]
                  }
                  name="role"
                  outsideClickIgnoreClass="ignore-react-onclickoutside"
                  placeHolder="Select role"
                  preventDefault={false}
                  required={true}
                  selectItem={[Function]}
                  stopPropagation={false}
                  text="Roles"
                >
                  <div>
                    <Component
                      name="role"
                      onChange={[Function]}
                      onClick={[Function]}
                      placeHolder="Select role"
                      required={true}
                      text="Roles"
                      type="text"
                      value=""
                    >
                      <Label
                        htmlFor="role"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": true,
                                "lastClassName": "hApbuX",
                                "rules": Array [
                                  "
    float: left;
    width: 25%;
    margin: 15px 5px;",
                                ],
                              },
                              "displayName": "Label",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "label",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          htmlFor="role"
                        >
                          <label
                            className="sc-htpNat hApbuX"
                            htmlFor="role"
                          >
                            Roles
                          </label>
                        </StyledComponent>
                      </Label>
                      <Input
                        autoComplete="off"
                        id="role"
                        name="role"
                        onChange={[Function]}
                        onClick={[Function]}
                        placeholder="Select role"
                        required={true}
                        type="text"
                        value=""
                      >
                        <StyledComponent
                          autoComplete="off"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bxivhb",
                                "isStatic": false,
                                "lastClassName": "byeNva",
                                "rules": Array [
                                  "
    margin-left: auto;
    margin-right: auto;
    width: 95%;
    padding: 12px;
    border: 1px solid #ccc;
    border-radius: 4px;
    resize: vertical;
    background-color: ",
                                  [Function],
                                  "
    display: ",
                                  [Function],
                                  "
    justify-content: ",
                                  [Function],
                                  "
    color: ",
                                  [Function],
                                  "
    ",
                                ],
                              },
                              "displayName": "Input",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bxivhb",
                              "target": "input",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          id="role"
                          name="role"
                          onChange={[Function]}
                          onClick={[Function]}
                          placeholder="Select role"
                          required={true}
                          type="text"
                          value=""
                        >
                          <input
                            autoComplete="off"
                            className="sc-bxivhb beFuQG"
                            id="role"
                            name="role"
                            onChange={[Function]}
                            onClick={[Function]}
                            placeholder="Select role"
                            required={true}
                            type="text"
                            value=""
                          />
                        </StyledComponent>
                      </Input>
                    </Component>
                  </div>
                </SelectBox>
              </OnClickOutside(SelectBox)>
              <OnClickOutside(SelectBox)
                eventTypes={
                  Array [
                    "mousedown",
                    "touchstart",
                  ]
                }
                excludeScrollbar={false}
                name="project"
                outsideClickIgnoreClass="ignore-react-onclickoutside"
                placeHolder="Select project"
                preventDefault={false}
                required={true}
                selectItem={[Function]}
                stopPropagation={false}
                text="Projects"
              >
                <SelectBox
                  disableOnClickOutside={[Function]}
                  enableOnClickOutside={[Function]}
                  eventTypes={
                    Array [
                      "mousedown",
                      "touchstart",
                    ]
                  }
                  name="project"
                  outsideClickIgnoreClass="ignore-react-onclickoutside"
                  placeHolder="Select project"
                  preventDefault={false}
                  required={true}
                  selectItem={[Function]}
                  stopPropagation={false}
                  text="Projects"
                >
                  <div>
                    <Component
                      name="project"
                      onChange={[Function]}
                      onClick={[Function]}
                      placeHolder="Select project"
                      required={true}
                      text="Projects"
                      type="text"
                      value=""
                    >
                      <Label
                        htmlFor="project"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-htpNat",
                                "isStatic": true,
                                "lastClassName": "hApbuX",
                                "rules": Array [
                                  "
    float: left;
    width: 25%;
    margin: 15px 5px;",
                                ],
                              },
                              "displayName": "Label",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-htpNat",
                              "target": "label",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          htmlFor="project"
                        >
                          <label
                            className="sc-htpNat hApbuX"
                            htmlFor="project"
                          >
                            Projects
                          </label>
                        </StyledComponent>
                      </Label>
                      <Input
                        autoComplete="off"
                        id="project"
                        name="project"
                        onChange={[Function]}
                        onClick={[Function]}
                        placeholder="Select project"
                        required={true}
                        type="text"
                        value=""
                      >
                        <StyledComponent
                          autoComplete="off"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bxivhb",
                                "isStatic": false,
                                "lastClassName": "byeNva",
                                "rules": Array [
                                  "
    margin-left: auto;
    margin-right: auto;
    width: 95%;
    padding: 12px;
    border: 1px solid #ccc;
    border-radius: 4px;
    resize: vertical;
    background-color: ",
                                  [Function],
                                  "
    display: ",
                                  [Function],
                                  "
    justify-content: ",
                                  [Function],
                                  "
    color: ",
                                  [Function],
                                  "
    ",
                                ],
                              },
                              "displayName": "Input",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bxivhb",
                              "target": "input",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          id="project"
                          name="project"
                          onChange={[Function]}
                          onClick={[Function]}
                          placeholder="Select project"
                          required={true}
                          type="text"
                          value=""
                        >
                          <input
                            autoComplete="off"
                            className="sc-bxivhb beFuQG"
                            id="project"
                            name="project"
                            onChange={[Function]}
                            onClick={[Function]}
                            placeholder="Select project"
                            required={true}
                            type="text"
                            value=""
                          />
                        </StyledComponent>
                      </Input>
                    </Component>
                  </div>
                </SelectBox>
              </OnClickOutside(SelectBox)>
              <Component
                type="submit"
                value="Submit role"
              >
                <Label>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": true,
                          "lastClassName": "hApbuX",
                          "rules": Array [
                            "
    float: left;
    width: 25%;
    margin: 15px 5px;",
                          ],
                        },
                        "displayName": "Label",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "label",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <label
                      className="sc-htpNat hApbuX"
                    />
                  </StyledComponent>
                </Label>
                <Input
                  autoComplete="off"
                  type="submit"
                  value="Submit role"
                >
                  <StyledComponent
                    autoComplete="off"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": false,
                          "lastClassName": "byeNva",
                          "rules": Array [
                            "
    margin-left: auto;
    margin-right: auto;
    width: 95%;
    padding: 12px;
    border: 1px solid #ccc;
    border-radius: 4px;
    resize: vertical;
    background-color: ",
                            [Function],
                            "
    display: ",
                            [Function],
                            "
    justify-content: ",
                            [Function],
                            "
    color: ",
                            [Function],
                            "
    ",
                          ],
                        },
                        "displayName": "Input",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "input",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    type="submit"
                    value="Submit role"
                  >
                    <input
                      autoComplete="off"
                      className="sc-bxivhb byeNva"
                      type="submit"
                      value="Submit role"
                    />
                  </StyledComponent>
                </Input>
              </Component>
            </form>
          </StyledComponent>
        </Form>
      </UserFormContainer>
    </Connect(UserFormContainer)>
  </UserForm>
</Provider>
`;
